src/sentry/static/sentry/app/views/eventsV2/resultsHeader.tsx

```diff
@@ -108,6 +108,7 @@ class ResultsHeader extends React.Component<Props, State> {
                 savedQuery={savedQuery}
                 savedQueryLoading={loading}
                 disabled={!hasFeature || (errorCode >= 400 && errorCode < 500)}
+                updateCallback={() => this.fetchData()}
```

Comment:
Would another approach be to update the conditions in `componentDidUpdate` to reload data when important parts of the saved query change? Or when the loading state goes from loading -> not loading?

Reply:
I looked at that but I couldn't find any robust way of detecting when it's being saved in `componentDidUpdate` through just the eventViews as the handlers don't expose any indications of the save button being clicked. The other alternative I was considering was to hoist the handlers up into the headers which might be closer to what you were thinking.

Reply:
Generally I'm a fan of hoisting actions up, but I don't think hoisting the action handlers would make this better as we'll need to drill a handlers down still.

Reply:
If the loading states don't let you handle the required logic in a robust way we should move forward with this.

Reply:
Saving and edits (like add a column) both just update the query params with no way of distinguishing between them. If we want to avoid having to pass down a handler entirely, the only other way I can think of would be to add the save state into the query params and parse that in `componentDidUpdate`.